edited the heroku section (#167)
datamade/how-to: heroku/deploy-a-django-app.md

heroku pipelines:connect ${APP_NAME} -r datamade/${APP_NAME}
```

Your CLI output should look like this:
```bash
heroku create ${APP_NAME}-staging -t datamade --manifest 
Reading heroku.yml manifest... done
Creating ⬢ demo-app-staging... done, stack is container
Adding heroku-postgresql... done
https://demo-app-staging.herokuapp.com/ | https://git.heroku.com/demo-app-staging.git

heroku pipelines:add ${APP_NAME}-staging -a ${APP_NAME}-staging -s staging
Adding ⬢ demo-app-staging to datamade-app pipeline as staging... done
```

If you would like to set up a production app as well, run the following commands
to create one and add it to your pipeline:

```bash
heroku create ${APP_NAME} -t datamade --manifest
heroku pipelines:add ${APP_NAME} -a ${APP_NAME} -s production
```

Once you have the environments you need, enable [review
apps](https://devcenter.heroku.com/articles/github-integration-review-apps)
for your pipeline:

```bash
# Note that these need to be two separate commands due to an open Heroku bug,
# since --autodeploy and --autodestroy require a PATCH request
heroku reviewapps:enable -p ${APP_NAME}
heroku reviewapps:enable -p ${APP_NAME} --autodeploy --autodestroy
```

> 

Reviewer: Can we add a heading here? Something like `Set configuration variables for review apps and deployments`? Be sure to add this to the contents at the top, as well.

Author: Breaks that section up nicely: 9c03a68

should be a string generated using the [XKCD password generator](https://preshing.com/20110811/xkcd-password-generator/),
and `DJANGO_ALLOWED_HOSTS` should be a comma-separated string of valid hosts
for your app.

```bash
heroku config:set -a ${APP_NAME}-staging DJANGO_SECRET_KEY=<random-string-here>
heroku config:set -a ${APP_NAME}-staging DJANGO_ALLOWED_HOSTS=.herokuapp.com

# Run these commands if you have a production application
heroku config:set -a ${APP_NAME} DJANGO_SECRET_KEY=<random-string-here>
heroku config:set -a ${APP_NAME} DJANGO_ALLOWED_HOSTS=<your-list-of-allowed-hosts-here>
```

Note that review app config vars cannot yet be set using the CLI, but you can set them in
the Heroku dashboard by navigating to the pipeline home page and visiting
`Settings > Review Apps > Review app config vars` in the nav.

Also note that while `DATABASE_URL` is probably required by your application, you don't actually
need to set it yourself. The Heroku Postgres add-on will [automatically define this
variable](https://devcenter.heroku.com/articles/heroku-postgresql#designating-a-primary-database)
when it provisions a database for your application.

> 

Reviewer: This could use a subhed, as well. `Configure deployments from Git branches`?

Author: This change is in the above commit, too: 9c03a68

}
```

## Set up GitHub Actions for CI

> For Heroku deployments, we use GitHub Actions instead of Travis for CI (continuous integration). Read the [how-to to set up GitHub Actions](https://github.com/datamade/how-to/blob/master/ci/github-actions.md).

Reviewer: ```suggestion
For Heroku deployments, we use GitHub Actions for CI (continuous integration). Read the [how-to to set up GitHub Actions](https://github.com/datamade/how-to/blob/master/ci/github-actions.md).
```

Author: Change is here: 2856c14

> ## Provision Heroku resources

Reviewer: You mentioned in the issue, and I agree, that increasingly, we most often come to these docs with our application already configured. For that reason, I think we should _start_ with provisioning Heroku resources, and move `Set up application code for Heroku` to the second to last section, as an optional step for legacy projects.

Author: Changed here: 44fd527. I added a little bit of context to intro text, too.

> Once your application is properly configured for Heroku, the following instructions
> will help you deploy your application to the platform.

Reviewer: ```suggestion
**If you initialized your application with DataMade's [`new-django-app` Cookiecutter template](https://github.com/datamade/how-to/tree/master/docker/templates)**, your app is already configured to run on Heroku.

**If you are migrating an existing app**, or if you want to learn more about the configuration that comes with the template, see [Set up application code for Heroku](#set-up-application-code-for-heroku), below, before proceeding with this step.

The following instructions will help you deploy your properly configured application to the platform.
```

### Install the Heroku CLI with the manifest plugin

> The fastest way to get a project up and running on Heroku is to use the [Heroku CLI](https://devcenter.heroku.com/articles/heroku-cli). Before you start, make sure you have the CLI installed locally. Once you install the CLI, you'll need to [switch over to the CLI's beta version](https://devcenter.heroku.com/articles/build-docker-images-heroku-yml#creating-your-app-from-setup). This allows you to use the `manifest` CLI plugin, which you must install:

Reviewer: ```suggestion
The fastest way to get a project up and running on Heroku is to use the [Heroku CLI](https://devcenter.heroku.com/articles/heroku-cli). Before you start, make sure you have the CLI installed locally. Once you install the CLI, you'll need to [switch over to the CLI's beta version](https://devcenter.heroku.com/articles/build-docker-images-heroku-yml#creating-your-app-from-setup). This allows you to use the `manifest` CLI plugin, which you must install in order to create your app and provision the appropriate add-ons (Postgres, etc.) from `heroku.yml`:
```

is not currently a way to set this type of configuration (and hence prevent these
kinds of emails being sent for review apps) because [Heroku defaults to the cheapest plan for
review app addons](https://help.heroku.com/SY28FR6H/why-aren-t-i-seeing-the-add-on-plan-specified-in-my-app-json-in-my-review-or-ci-app).

> ### Help! My staging pipeline doesn't have a database!

Reviewer: Nice.

You might deploy a review app and everything works. Then you merge your code to master, which builds a new version to the staging pipeline. But for some reason, there is no database provisioned for staging.

Did you have the `manifest` CLI plugin installed when you first created the Heroku pipeline? If not, then it won't provision the Postgres add-on. [See this step](#install-the-heroku-cli-with-the-manifest-plugin).

> Here's an example where the `manifest` plugin **was not installed** when creating an app:

Reviewer: I think it would be more useful to include the expected output in the section where we create the apps, that way we can catch this error early.

Author: Good idea.

Author: Included it here too: c4c7a7a.
